design/cluster-scope-resource-filter.md

```diff
@@ -0,0 +1,394 @@
+			- [all namespaced resources + no cluster resources](#all-namespaced-resources--no-cluster-resources)
+			- [all namespaced resources + only related cluster resources](#all-namespaced-resources--only-related-cluster-resources)
+			- [all namespaced resources + some additional cluster resources](#all-namespaced-resources--some-additional-cluster-resources)
+			- [all namespaced resources + all cluster resources](#all-namespaced-resources--all-cluster-resources)
+			- [describe command change](#describe-command-change)
+	- [Detailed Design](#detailed-design)
+	- [Alternatives Considered](#alternatives-considered)
+	- [Security Considerations](#security-considerations)
+	- [Compatibility](#compatibility)
+	- [Implementation](#implementation)
+	- [Open Issues](#open-issues)
+
+## Abstract
+The current filter (IncludedResources/ExcludedResources + IncludeClusterResources flag) is not enough for some special cases, e.g. all namespaced resources + some kind of cluster resource and all namespaced resources + cluster resource excludes.
+Propose to add a new group of resource filtering parameters, which can distinguish cluster and namespaced resources.
+
+## Background
+There are two sets of resource filters for Velero: `IncludedNamespaces/ExcludedNamespaces` and `IncludedResources/ExcludedResources`. 
+`IncludedResources` means only including the resource types specified in the parameter. Both cluster and namespaced resources are handled in this parameter by now.
+The k8s resources are separated into cluster scope and namespaced scope.
+As a result, it's hard to include all resources in one group and only including specified resource in the other group.
+
+## Goals
+- Make Velero can support more complicated namespaced and cluster resources filtering scenarios in backup.
+
+## Non Goals
+- Enrich the resource filtering rules, for example, advanced PV filtering and filtering by resource names.
+
+
+## High-Level Design
+Four new parameters are added into command `velero backup create`: `--include-cluster-scope-resources`, `--exclude-cluster-scope-resources`, `--include-namespaced-resources` and `--exclude-namespaced-resources`. 
+`--include-cluster-scope-resources` and `--exclude-cluster-scope-resources` are used to filter cluster scope resources included or excluded in backup per resource type.
+`--include-namespaced-resources` and `--exclude-namespaced-resources` are used to filter namespace scope resources included or excluded in backup per resource type.
+Restore and other code pieces also use resource filtering will be handled in future releases.
+
+### Parameters Rules
+
+* `--include-namespaced-resources` valid value is comma separated string. Not setting the parameter means including all namespace resources.
+
+* `--include-cluster-scope-resources`, `--exclude-cluster-scope-resources` and `--exclude-namespaced-resources` valid value include `*` and comma separated string.
```

Comment:
1. Shall we enforce that the format has to be `$resource.$group`?  As I recall this is more consistent with existing behavior.
2. shall we allow the values to contain wildcard in the string items, like `--exclude-namespaced-resources=*.velero.io`

Reply:
Just confirmed. `$kind-name`, `$kind-name.$group-name`, and capital or small all work, for example, `storageClass.storage.k8s.io, servicE, configMap`.
And because this is a rather flexible parameter, working out a regular expression may not be easy. I will try to implement it, but if that consumes effort, may not do it in this release.

Reply:
Currently wildcard is only really supported in includes, and it's only `*` for all -- I don't think `*.velero.io` is supported on include or exclude. We could consider this sort of flexible wildcard support as a follow-on enhancement, since it's not supported in the current implementation either.

Reply:
Agreed. I will check whether this is doable and the effort. May not do it this time.